webapp/src/test/java/org/apache/falcon/resource/EntityManagerJerseyIT.java

```diff
@@ -198,6 +198,8 @@ public void testOptionalInput() throws Exception {
         File tmpFile = TestContext.getTempFile();
         EntityType.PROCESS.getMarshaller().marshal(process, tmpFile);
         schedule(context);
+        waitForStatus(EntityType.PROCESS.name(), context.getProcessName(), START_INSTANCE,
```

Comment:
One case where docs are ahead of code. The Entity Specification already says this -
"Optional Inputs

User can mention one or more inputs as optional inputs. In such cases the job does not wait on those inputs which are mentioned as optional. If they are present it considers them otherwise continue with the compulsory ones. " - http://falcon.apache.org/EntitySpecification.html#Process_Specification

Let me know if I should add more.